src/wp-includes/webfonts-api/providers/class-wp-webfonts-provider.php

```diff
@@ -0,0 +1,200 @@
+	 *
+	 *      @type string $key => @type array resource hint.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @var array
+	 */
+	protected $resource_hints = array();
+
+	/**
+	 * Whether the provider fetches external resources or not.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @var bool
+	 */
+	protected $is_external = true;
+
+	/**
+	 * Get the provider's unique ID.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @return string
+	 */
+	public function get_id() {
+		return $this->id;
+	}
+
+	/**
+	 * Get the root URL for the provider.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @return string
+	 */
+	public function get_root_url() {
+		return $this->root_url;
+	}
+
+	/**
+	 * Sets this provider's webfonts property.
+	 *
+	 * The API's Controller passes this provider's webfonts
+	 * for processing here in the provider.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param array[] $webfonts Registered webfonts.
+	 */
+	public function set_webfonts( array $webfonts ) {
+		$this->webfonts = $webfonts;
+	}
+
+	/**
+	 * Get the CSS for the font.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @return string Webfonts CSS.
+	 */
+	abstract public function get_css();
+
+	/**
+	 * Get cached styles from a remote URL.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param string $id               An ID used to cache the styles.
+	 * @param string $url              The URL to fetch.
+	 * @param array  $args             The arguments to pass to wp_remote_get().
+	 * @param array  $additional_props Additional properties to add to the @font-face styles.
+	 * @return string The styles.
+	 */
+	public function get_cached_remote_styles( $id, $url, array $args = array(), array $additional_props = array() ) {
+		$css = get_site_transient( $id );
+
+		// Get remote response and cache the CSS if it hasn't been cached already.
+		if ( false === $css ) {
+			$css = $this->get_remote_styles( $url, $args );
+
+			// Early return if the request failed.
+			// Cache an empty string for 60 seconds to avoid bottlenecks.
+			if ( empty( $css ) ) {
+				set_site_transient( $id, '', 60 );
+				return '';
+			}
+
+			// Cache the CSS for a month.
+			set_site_transient( $id, $css, MONTH_IN_SECONDS );
+		}
+
+		// If there are additional props not included in the CSS provided by the API, add them to the final CSS.
+		foreach ( $additional_props as $prop ) {
+			$css = str_replace(
+				'@font-face {',
+				'@font-face {' . $prop . ':' . $this->params[ $prop ] . ';',
+				$css
+			);
+		}
+
+		return $css;
+	}
+
+	/**
+	 * Get styles from a remote URL.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param string $url  The URL to fetch.
+	 * @param array  $args The arguments to pass to wp_remote_get().
+	 * @return string The styles on success. Empty string on failure.
+	 */
+	public function get_remote_styles( $url, array $args = array() ) {
+		// Use a modern user-agent, to get woff2 files.
+		$args['user-agent'] = 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:73.0) Gecko/20100101 Firefox/73.0';
+
+		// Get the remote URL contents.
+		$response = wp_remote_get( $url, $args );
```

Comment:
Would be helpful to have a filter for `args`  ?

Reply:
@mitogh do you foresee a need of themes and/or plugins to modify these arguments for a specific font service provider? For example, do you think a theme or plugin needs access to modify these arguments for fetching from the Google Fonts API?

Reply:
Let's say locally I want to load a file via `https` but I don't have the certificate I would like to set `sslverify` to `false` another use case might be extend the default timeout of the request specify specific headers.

Reply:
The args for filterable via `'http_request_args'` within the `wp_remote_get()` function. Would this filter fit your needs?

Reply:
Yep I think that would work :smile: